html/modules/custom/reliefweb_import/src/Service/WorkDayJobImporter.php

```diff
@@ -0,0 +1,384 @@
+
+  /**
+   * The settings for the WorkDay service.
+   *
+   * @var array
+   */
+  protected array $settings;
+
+  /**
+   * Set the settings for the WorkDay service.
+   *
+   * @param array $settings
+   *   The settings to use for the WorkDay service.
+   */
+  public function setSettings(array $settings): void {
+    $this->settings = $settings;
+  }
+
+  /**
+   * Validate settings.
+   */
+  public function validateSettings(): void {
+    $required_settings = [
+      'base_url',
+      'tenant',
+      'client_id',
+      'client_secret',
+      'refresh_token',
+      'source_id',
+      'uid',
+    ];
+
+    $missing_settings = [];
+    foreach ($required_settings as $setting) {
+      if (empty($this->settings[$setting])) {
+        $missing_settings[] = $setting;
+      }
+    }
+
+    if (!empty($missing_settings)) {
+      throw new \InvalidArgumentException('Missing required WorkDay settings: ' . implode(', ', $missing_settings));
+    }
+
+    // Make sure base_url is a valid URL.
+    if (filter_var($this->settings['base_url'], FILTER_VALIDATE_URL) === FALSE) {
+      throw new \InvalidArgumentException('The base_url setting is not a valid URL.');
+    }
+
+    // Make sure uid points to an existing user.
+    if (!$this->entityTypeManager->getStorage('user')->load($this->settings['uid'])) {
+      throw new \InvalidArgumentException('The uid setting is not valid.');
+    }
+
+    // Make sure source_id points to an existing source.
+    if (!$this->entityTypeManager->getStorage('taxonomy_term')->load($this->settings['source_id'])) {
+      throw new \InvalidArgumentException('The source_id setting is not valid.');
+    }
+  }
+
+  /**
+   * Get authorization token from WorkDay.
+   */
+  public function getAuthToken(): string {
+    $timeout = $this->settings['timeout'] ?? 10;
+    $base_url = rtrim($this->settings['base_url'] ?? '', '/');
+    $tenant = $this->settings['tenant'] ?? '';
+    $client_id = $this->settings['client_id'] ?? '';
+    $client_secret = $this->settings['client_secret'] ?? '';
+    $refresh_token = $this->settings['refresh_token'] ?? '';
+    $token_url = $base_url . '/ccx/oauth2/' . $tenant . '/token';
+
+    $response = $this->httpClient->request('POST', $token_url, [
+      'connect_timeout' => $timeout,
+      'timeout' => $timeout,
+      'headers' => [
+        'Content-Type' => 'application/x-www-form-urlencoded',
+        'Accept' => 'application/json',
+      ],
+      'form_params' => [
+        'grant_type' => 'refresh_token',
+        'client_id' => $client_id,
+        'client_secret' => $client_secret,
+        'refresh_token' => $refresh_token,
+      ],
+    ]);
+
+    if ($response->getStatusCode() !== 200) {
+      throw new \Exception('Failure with response code: ' . $response->getStatusCode());
+    }
+
+    try {
+      $token = json_decode($response->getBody()->getContents(), TRUE, flags: \JSON_THROW_ON_ERROR)['access_token'];
```

Comment:
```suggestion
      $content = json_decode($response->getBody()->getContents(), TRUE, flags: \JSON_THROW_ON_ERROR);
      if (!is_array($content) || !isset($content['access_token'])) {
         throw new \Exception('Missing access token.');
      }
```

If `json_decode($response->getBody()->getContents(), TRUE, flags: \JSON_THROW_ON_ERROR)` doesn't return an array with the `access_token` key, PHP is going to issue an undefined property notice/warning.